src/app/components/toggle-button/toggle-button.component.ts

```diff
@@ -0,0 +1,19 @@
+import { Component } from '@angular/core';
+
+@Component({
+  selector: 'app-toggle-button',
+  templateUrl: './toggle-button.component.html',
+  styleUrl: './toggle-button.component.scss'
+})
+export class ToggleButtonComponent {
+  active:boolean = false;
+
+  changeColor():void{
+    if(this.active){
```

Comment:
todo este bloque de código (12-18), puede cambiarse por un oneliner así:
this.active = !this.active;